From 2318f056a6c740f66fdeb6b9f47cea319dcdeb20 Mon Sep 17 00:00:00 2001
From: Dom Cobley <popcornmix@gmail.com>
Date: Thu, 2 Jul 2020 19:36:08 +0100
Subject: [PATCH] dts: Enable NO_WAIT_RESP for hdmi audio dma

Without this set, DVP_CFG_MAI0_CTL indicates occasional
DLATE errors when configured to 8 channel 192kHz

Signed-off-by: Dom Cobley <popcornmix@gmail.com>
---
 arch/arm/boot/dts/bcm2711-rpi.dtsi    | 4 ++--
 arch/arm/boot/dts/bcm2835-common.dtsi | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

--- a/arch/arm/boot/dts/bcm2711-rpi.dtsi
+++ b/arch/arm/boot/dts/bcm2711-rpi.dtsi
@@ -74,7 +74,7 @@
 			clock-names = "hdmi";
 			resets = <&dvp 0>;
 			ddc = <&ddc0>;
-			dmas = <&dma 10>;
+			dmas = <&dma (10|(1<<27))>;
 			dma-names = "audio-rx";
 			interrupts = <GIC_SPI 96 IRQ_TYPE_LEVEL_HIGH>;
 			status = "disabled";
@@ -114,7 +114,7 @@
 			clocks = <&firmware_clocks 13>;
 			clock-names = "hdmi";
 			resets = <&dvp 1>;
-			dmas = <&dma 17>;
+			dmas = <&dma (17|(1<<27))>;
 			dma-names = "audio-rx";
 			interrupts = <GIC_SPI 96 IRQ_TYPE_LEVEL_HIGH>;
 			status = "disabled";
--- a/arch/arm/boot/dts/bcm2835-common.dtsi
+++ b/arch/arm/boot/dts/bcm2835-common.dtsi
@@ -117,7 +117,7 @@
 			clocks = <&clocks BCM2835_PLLH_PIX>,
 				 <&clocks BCM2835_CLOCK_HSM>;
 			clock-names = "pixel", "hdmi";
-			dmas = <&dma 17>;
+			dmas = <&dma (17|(1<<27))>;
 			dma-names = "audio-rx";
 			status = "disabled";
 		};
